eth_account/hdaccount/mnemonic.py

```diff
@@ -0,0 +1,205 @@
+# Originally from: https://github.com/trezor/python-mnemonic
+#
+# Copyright (c) 2013 Pavol Rusnak
+# Copyright (c) 2017 mruddy
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy of
+# this software and associated documentation files (the "Software"), to deal in
+# the Software without restriction, including without limitation the rights to
+# use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies
+# of the Software, and to permit persons to whom the Software is furnished to do
+# so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in all
+# copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY,
+# WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+# CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+#
+
+import os
+from pathlib import (
+    Path,
+)
+import secrets
+
+from bitarray import (
+    bitarray,
+)
+from bitarray.util import (
+    ba2int,
+    int2ba,
+)
+from eth_utils import (
+    ValidationError,
+)
+
+from ._utils import (
+    normalize_string,
+    pbkdf2_hmac_sha512,
+    sha256,
+)
+
+VALID_ENTROPY_SIZES = [16, 20, 24, 28, 32]
+VALID_WORD_COUNTS = [12, 15, 18, 21, 24]
+WORDLIST_DIR = Path(__file__).parent / "wordlist"
+WORDLIST_LEN = 2048
+
+_cached_wordlists = dict()
+
+
+def get_wordlist(language):
+    if language in _cached_wordlists.keys():
+        return _cached_wordlists[language]
+    with open(WORDLIST_DIR / f"{language}.txt", "r", encoding="utf-8") as f:
+        wordlist = [w.strip() for w in f.readlines()]
+    if len(wordlist) != WORDLIST_LEN:
+        raise ValidationError(
+            f"Wordlist should contain {WORDLIST_LEN} words, "
+            f"but it contains {len(wordlist)} words."
+        )
+    _cached_wordlists[language] = wordlist
+    return wordlist
+
+
+class Mnemonic:
+    def __init__(self, language):
+        if language not in Mnemonic.list_languages():
```

Comment:
I'm still seeing the AttributeError instead of the expected ValidationError. I believe you need something like:

```python
language_list = Mnemonic.list_language()
if language not in language_list:
    raise ValidationError(
        f'Invalid language choice "{language}", must be one of {language_list}'
    ) 
```

Reply:
Hmm, I'm not sure what you're seeing, but I restructured it a bit and tested your scenario, so it shouldn't be happening anymore.